### PR TITLE
CTDA-1373 Support IE917 message type

### PR DESCRIPTION
**Pull request by DavidGregory084**
TODO:

- [X] Figure out what to do about MessageSummary
- [X] Check the new transitions are correct

### CODE REVIEW

**Comment by DavidGregory084** [comment]
![New status transitions](https://dreampuf.github.io/GraphvizOnline/#digraph%20%22Departure%20Status%20Transitions%22%20%7B%0A%09margin%20%3D%20%220%22%0A%09page%20%3D%20%228.5%2C11.0%22%0A%09size%20%3D%20%228.5%2C11.0%22%0A%09rotate%20%3D%20%220%22%0A%09ratio%20%3D%20%22fill%22%0A%09%22DepartureSubmitted%22%20-%3E%20%22DepartureRejected%22%0A%09%22DepartureSubmitted%22%20-%3E%20%22MrnAllocated%22%0A%09%22DepartureSubmitted%22%20-%3E%20%22PositiveAcknowledgement%22%0A%09%22DepartureSubmitted%22%20-%3E%20%22DepartureSubmittedNegativeAcknowledgement%22%0A%09%22GuaranteeNotValid%22%20-%3E%20%22DeclarationCancellationRequest%22%0A%09%22GuaranteeNotValid%22%20-%3E%20%22ReleaseForTransit%22%0A%09%22GuaranteeNotValid%22%20-%3E%20%22NoReleaseForTransit%22%0A%09%22ReleaseForTransit%22%20-%3E%20%22DeclarationCancellationRequest%22%0A%09%22ReleaseForTransit%22%20-%3E%20%22WriteOffNotification%22%0A%09%22ReleaseForTransit%22%20-%3E%20%22CancellationDecision%22%0A%09%22PositiveAcknowledgement%22%20-%3E%20%22MrnAllocated%22%0A%09%22DeclarationCancellationRequestNegativeAcknowledgement%22%20-%3E%20%22DeclarationCancellationRequest%22%0A%09%22MrnAllocated%22%20-%3E%20%22GuaranteeNotValid%22%0A%09%22MrnAllocated%22%20-%3E%20%22ReleaseForTransit%22%0A%09%22MrnAllocated%22%20-%3E%20%22NoReleaseForTransit%22%0A%09%22MrnAllocated%22%20-%3E%20%22DeclarationCancellationRequest%22%0A%09%22MrnAllocated%22%20-%3E%20%22ControlDecisionNotification%22%0A%09%22Initialized%22%20-%3E%20%22DepartureSubmitted%22%0A%09%22Initialized%22%20-%3E%20%22PositiveAcknowledgement%22%0A%09%22Initialized%22%20-%3E%20%22DepartureSubmittedNegativeAcknowledgement%22%0A%09%22Initialized%22%20-%3E%20%22MrnAllocated%22%0A%09%22Initialized%22%20-%3E%20%22DepartureRejected%22%0A%09%22DeclarationCancellationRequest%22%20-%3E%20%22DeclarationCancellationRequestNegativeAcknowledgement%22%0A%09%22DeclarationCancellationRequest%22%20-%3E%20%22CancellationDecision%22%0A%09%22CancellationDecision%22%20-%3E%20%22WriteOffNotification%22%0A%09%22ControlDecisionNotification%22%20-%3E%20%22ReleaseForTransit%22%0A%09%22ControlDecisionNotification%22%20-%3E%20%22NoReleaseForTransit%22%0A%7D)

**Review by crystalmark — APPROVED** [review]

**Review by crystalmark — APPROVED** [review]

File: app/models/DepartureStatus.scala [hunk 1]
```diff
@@ -46,6 +46,10 @@ object DepartureStatus extends Enumerable.Implicits with MongoDateTimeFormats {
 
   case object GuaranteeNotValid extends DepartureStatus
 
+  case object DepartureSubmittedNegativeAcknowledgement extends DepartureStatus
+
+  case object DeclarationCancellationRequestNegativeAcknowledgement extends DepartureStatus
```
Comment (amaalali) [inline app/models/DepartureStatus.scala:51]: @ashwinirudrappa for your info

File: app/models/MessagesSummary.scala [hunk 2]
```diff
@@ -18,31 +18,37 @@ package models
...
+  declaration: MessageId,
+  declarationRejection: Option[MessageId] = None,
+  mrnAllocated: Option[MessageId] = None,
+  guaranteeNotValid: Option[MessageId] = None,
+  cancellationDecision: Option[MessageId] = None,
+  declarationCancellationRequest: Option[MessageId] = None,
+  noReleaseForTransit: Option[MessageId] = None,
+  controlDecision: Option[MessageId] = None,
+  releaseForTransit: Option[MessageId] = None,
+  xmlSubmissionNegativeAcknowledgement: Option[MessageId] = None
```
Comment (amaalali) [inline app/models/MessagesSummary.scala:32]: 🙌

**Review by amaalali — APPROVED** [review]
This looks good to me. @ashwinirudrappa Can you do a quick test with the FE?

**Pull request merged**